sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2538,6 +2538,56 @@ case class Encode(value: Expression, charset: Expression)
     newLeft: Expression, newRight: Expression): Encode = copy(value = newLeft, charset = newRight)
 }
 
+/**
+ * Converts the input expression to a binary value based on the supplied format.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(str[, fmt]) - Converts the input `str` to a binary value based on the supplied `fmt`.
+      By default, the binary format for conversion is "hex" if `fmt` is omitted.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('abc', 'utf-8');
+       abc
+  """,
+  since = "3.3.0",
+  group = "string_funcs")
+// scalastyle:on line.size.limit
+case class ToBinary(expr: Expression, format: Option[Expression], child: Expression)
+  extends RuntimeReplaceable {
+
+  def this(expr: Expression, format: Expression) = this(expr, Option(format),
+    format.toString match {
+      case "hex" => Unhex(expr)
+      case "utf-8" => Encode(expr, Literal("UTF-8"))
+      case "base64" => UnBase64(expr)
+      case "base2" => Cast(expr, BinaryType)
+    }
+  )
+
+  def this(expr: Expression) = this(expr, None, Unhex(expr))
+
+  override def flatArguments: Iterator[Any] = Iterator(expr, format)
+  override def exprsReplaced: Seq[Expression] = expr +: format.toSeq
+
+  override def prettyName: String = "to_binary"
+  override def dataType: DataType = BinaryType
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (format.isEmpty || Seq(Literal("hex"), Literal("utf-8"), Literal("base64"), Literal("base2"))
+      .contains(format.get)) {
+      super.checkInputDataTypes()
+    } else {
+      TypeCheckResult.TypeCheckFailure(s"Unsupported encoding format")
```

Comment:
Maybe we should also print out which argument was given. e.g.)

```scala
      TypeCheckResult.TypeCheckFailure(
        s"Unsupported encoding format: $format. " +
          "The format has to be a string literal of 'hex', ...")
```

Reply:
Updated, now the exception looks like:
```
scala> spark.sql("select to_binary('abc', 'a')").show()
org.apache.spark.sql.AnalysisException: cannot resolve 'to_binary('abc', 'a')' due to data type mismatch: Unsupported encoding format: Some(a). The format has to be a string literal of 'hex', 'utf-8', 'base2', or 'base64'
...
```